client/worker.go

```diff
@@ -54,56 +61,46 @@ func newWorker(client *http.Client, endpoint string, accessToken string, disable
 }
 
 func (w *worker) export(ctx context.Context, batches []*jaegerpb.Batch, accessToken string) *ErrSend {
-	ctx, span := trace.StartSpan(ctx, "export")
+	ctx, span := w.tracer.Start(ctx, "export")
 	defer span.End()
 
 	var spansCount int
 	for _, batch := range batches {
 		spansCount += len(batch.Spans)
 	}
 
-	span.AddAttributes(trace.Int64Attribute("spans", int64(spansCount)))
-	span.AddAttributes(trace.Int64Attribute("batches", int64(len(batches))))
+	span.SetAttributes(attribute.Int64("spans", int64(spansCount)))
+	span.SetAttributes(attribute.Int64("batches", int64(len(batches))))
+
 	if spansCount == 0 {
 		return nil
 	}
 
-	sr, err := w.prepare(ctx, batches, spansCount)
+	sr, err := w.prepare(batches, spansCount)
 	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
 		recordEncodingFailure(ctx, sr)
-		span.SetStatus(trace.Status{
-			Code: trace.StatusCodeInvalidArgument,
-		})
-		return &ErrSend{Err: err}
+		return &ErrSend{Err: err, Permanent: true}
 	}
 
 	serr := w.send(ctx, sr, accessToken)
-
 	if serr == nil {
 		recordSuccess(ctx, sr)
 		return nil
 	}
 	recordSendFailure(ctx, sr)
-	span.SetStatus(trace.Status{
-		Code: OCStatusCodeFromHTTP(int32(serr.StatusCode)),
-	})
 
+	span.SetStatus(codes.Error, serr.Err.Error())
 	if serr.Permanent {
 		recordDrops(ctx, sr)
+		span.SetStatus(codes.Error, "unable to prepare request")
```

Comment:
Is this correct? `serr` is error from `send` not `prepare`

Reply:
You are correct, and anyway we set the error 2 lines above. Removing.